src/core/server/plugins/plugins_service.ts

```diff
@@ -271,10 +271,10 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
       PluginName,
       { plugin: PluginWrapper; isEnabled: boolean }
     >();
-    await plugin$
+    // 1. Register config descriptors and deprecations
+    const plugins = await plugin$
       .pipe(
-        // 1. Register config descriptors and deprecations
-        map((plugin) => {
+        tap((plugin) => {
```

Comment:
optional: I would either do it rxjs all-the-way, or not at all

Either something like:

```ts
plugin$
   .pipe(tap(...))
   .toArray()
   .concatMap((plugins) => of(plugins))
   .concatMap((plugin) => ...);
```

or

```ts
const plugins = await plugin$.pipe(toArray()).toPromise();
for (const plugin of plugins) { /* loop 1 */ }
for (const plugin of plugins) { /* loop 2 */}
```

But current implem is also totally fine.